### PR TITLE
Add iterable support

### PR DESCRIPTION
**Pull request by falsandtru**

### CODE REVIEW

File: implementation.js [hunk 1]
```diff
@@ -1,8 +1,43 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var global = require('system.global')();
+var isCallable = require('is-callable');
+var iterator = typeof Symbol === 'function' ? Symbol.iterator : '@@iterator';
```
Comment (ljharb) [inline implementation.js]: this is not sufficient. we can only use the string form when that's what `for..of` uses. In addition, there are some browsers that have NEITHER of these ways to determine iterability, and only piping through `for..of` can detect that.

File: implementation.js [hunk 2]
```diff
@@ -1,8 +1,43 @@
...
 
 /*! https://mths.be/array-from v0.2.0 by @mathias */
+var parse = function (items) {
+	if (isIterable(items)) {
+		return fromIterable(items);
+	}
+	if (typeof items === 'string') {
+		return items.match(/[\uD800-\uDBFF][\uDC00-\uDFFF]?|[^\uD800-\uDFFF]|./g) || [];
+	}
+	if (global.Map && items instanceof global.Map && isCallable(items.entries)) {
```
Comment (ljharb) [inline implementation.js]: `instanceof` will not work cross-realm (like iframes etc). We can't rely on anything using `instanceof` of `===`.

**Comment by ljharb** [comment]
This in fact omits many of the *absolutely utterly necessary* things that were added in response to my comments on #27. I'm happy to keep this open as a parallel proposal, but iteration is complicated, and it simply can not be done without gobs and gobs of code.

File: implementation.js (hunk 2 above)
Comment (falsandtru) [inline implementation.js]: Right, I'll fix it.

File: implementation.js (hunk 1 above)
Comment (falsandtru) [inline implementation.js]: Sorry, I don't know that well.

**Comment by falsandtru** [comment]
I created this pr for quick update of polyfill-service because I need this support. Can this code contribute   quick update?

**Comment by ljharb** [comment]
There is no quick update that can reliably provide iterator support here, and partial support is worse than none.

**Comment by falsandtru** [comment]
I see, then I stop this pr. What is needed in #27 for iterator support?

**Pull request closed without merging**